Subject: libtool.m4: build shared libraries on substrate

libtool's host_os dispatch has no substrate arm, so on substrate it
decides shared libraries are unsupported and --enable-shared quietly
yields static archives only.  Substrate is an ELF/ld.so system with the
same SONAME and PIC conventions as GNU/Linux: fold substrate* into the
linux* arms for the dynamic linker, deplibs check, PIC flags (C and
C++) and shared-library link commands.  This is the macro that every
package autoreconf'd on substrate picks up.

---
--- a/m4/libtool.m4	2026-09-27 23:53:07.002637532 -0500
+++ b/m4/libtool.m4	2026-09-27 23:53:16.168536633 -0500
@@ -3047,7 +3047,7 @@
   ;;
 
 # This must be glibc/ELF.
-linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
+linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu* | substrate*)
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
   need_version=no
@@ -3802,7 +3802,7 @@
   ;;
 
 # This must be glibc/ELF.
-linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
+linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu* | substrate*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -4629,7 +4629,7 @@
 	;;
       *-mlibc)
 	;;
-      linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
+      linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu* | substrate*)
 	case $cc_basename in
 	  KCC*)
 	    # KAI C++ Compiler
@@ -4961,7 +4961,7 @@
       _LT_TAGVAR(lt_prog_compiler_static, $1)='-static'
       ;;
 
-    linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu* | freebsd*)
+    linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu* | freebsd* | substrate*)
       case $cc_basename in
       # old Intel for x86_64, which still supported -KPIC.
       ecc*)
@@ -5469,7 +5469,7 @@
 	_LT_TAGVAR(archive_expsym_cmds, $1)='$CC -shared $pic_flag $libobjs $deplibs $compiler_flags $wl-soname $wl$soname $wl-retain-symbols-file $wl$export_symbols -o $lib'
       ;;
 
-    gnu* | linux* | tpf* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu)
+    gnu* | linux* | tpf* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | substrate*)
       tmp_diet=no
       if test linux-dietlibc = "$host_os"; then
 	case $cc_basename in
@@ -6090,7 +6090,7 @@
     *-mlibc)
       ;;
 
-    linux*)
+    linux* | substrate*)
       case $cc_basename in
       tcc*)
 	# Fabrice Bellard et al's Tiny C Compiler
@@ -7247,7 +7247,7 @@
         _LT_TAGVAR(ld_shlibs, $1)=yes
 	;;
 
-      linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
+      linux* | ironclad*-gnu | k*bsd*-gnu | kopensolaris*-gnu | gnu* | substrate*)
         case $cc_basename in
           KCC*)
 	    # Kuck and Associates, Inc. (KAI) C++ Compiler
